Rooms/masterBedroom.json

```diff
@@ -4,14 +4,21 @@
     "shortDescription": "You're in the Master Bedroom. Your bed is on the floor. There's a light switch and closet nearby.",
     "lockRoom": "False",
     "conditionals": {
-        "boxes": {
+        "boxesRoom": {
```

Comment:
I don't understand why we need both "boxesRoom" and "boxes". Also, I don't think "boxesRoom" will ever be reached, because conditionals are only accessed when the user types in their name (or an alias we have assigned to it in itemDictionary.json) and I don't think it will be possible to differentiate between "boxes" (feature) and "boxes" (room).

Reply:
Forgive me if I'm over- or under-explaining here, but I can try.

"Room" conditions and "feature" conditions are accessed and processed slightly differently, although they progress in the same way (through "steps").

The main problem I faced (which this PR solves) was deciding whether to read "room" and "feature" conditions into two different JSON dicts, or the same dict with category labels.

"Room" conditions are written to be part of a room's description (called using "look" or by entering the room). I consider this to be a more generalized, third person perspective when writing the story content. "Feature" conditions, on the other hand, are written to be a direct response to the player's actions - a perspective that offers richer detail and possibly includes an action the player took. This is one reason why they're two separate categories. While the boxes here aren't necessarily the best example, this would be a better one:

- Chest in a "room" description (called by "look") might be described as "There's a wooden chest in the corner."
- The same chest in a "feature" description (as a response to "look at chest") might be more like "You run your hands over the rough wood and metal detailing. There are scratches in the top and a sturdy padlock on the front."

So while it's _technically possible_ to just make them both say "There's a wooden chest in the corner," it doesn't offer quite as much narrative richness to the interaction, in my opinion. I think it sounds better, after I've been playing around with the game. If the player's response to the "look at" command is the exact same as the "look" description, it largely defeats the purpose of having a separate command. If I were to change anything here in this PR, I'd maybe just make the "feature" boxes description differ more from the "room" version - but I'm less concerned about those details at this moment.

The bottom line being: "feature" descriptions are not always something you want included in a room's main description, and "room" descriptions are not always something you want to receive in response to a player's command.

As for whether or not it they all can be reached, I've made sure they do indeed all work. This is because the "name" (key) of the "room" conditions isn't actually used for processing the condition at all. It only exists to create the condition in the first place. When any of the conditions are triggered, the function only looks at the condition's trigger/trigger sequence (the target called by the player's input command).

It basically works like this:
- Player directly triggers "room" conditions (something in the room that can be revealed, like the box contents)
- Room class unlocks any relevant items within the room (makes the battery in the box accessible to player)
- Conditional class unlocks any other "features" that are revealed by the action (like the clock and clothes)
- Conditional class increments all relevant "room" or "feature" conditions to the next step, if applicable

A "feature" condition's name attribute is needed to retrieve a description in response to a command (through the Room.verbResponses method). A "room" condition's description is automatically included when pulling a room's long or short description (through the Room.getConditionalDesc method). Both of these methods check the condition's type and only retrieve the appropriate description for that situation, to differentiate them.

Because the methods check the condition's type, theoretically you could have both the "room" boxes and "feature" boxes have the same name ("boxes"). But since the "room" conditions' name isn't needed anyway, I changed it to "boxesRoom" to be more explicitly different than the "feature" boxes.

Additionally, only the "feature" conditions are passed through the verbInteractions dict. What this means in practice is that while right now we're talking about the effect of calling "look at" on the boxes, and "look at" is very similar to "look", there are also other cases where you would definitely want "use" or "eat" or "open" to have a very different response than you would when calling "look" or "look at."

I understand it feels complicated, but I am mostly just finding out that it IS complicated to handle situational gameplay in a natural-feeling way. Like with everything else, I'm open to other options if needed. This one does work though.

![boxes](https://user-images.githubusercontent.com/36237047/201755552-2df23ade-e326-4c3b-b534-25c66c271dc0.png)

Reply:
Ohhh, ok. Gotcha. Thank you for the detailed explanation, this helped clear some things up for me!